tests/unit/src/main/scala/scala/meta/internal/metals/debug/TestDebugger.scala

```diff
@@ -65,7 +65,15 @@ final class TestDebugger(
       .map { response =>
         // the breakpoint notification we receive does not contain the source
         // hence we have to register breakpoints here
-        response.getBreakpoints.foreach(this.breakpoints.register)
+        response.getBreakpoints.foreach { brPoint =>
+          // note(@tgodzik) from time to time breakpoints are sent back without the source,
```

Comment:
This should help with the flaky tests better, since we are not sure how it is possible that source is very rarely null.